modules/a1MediaRtdProvider.js

```diff
@@ -0,0 +1,74 @@
+import { MODULE_TYPE_RTD } from '../src/activities/modules.js';
+import { loadExternalScript } from '../src/adloader.js';
+import { submodule } from '../src/hook.js';
+import { getStorageManager } from '../src/storageManager.js';
+import { isEmptyStr, mergeDeep } from '../src/utils.js';
+
+const REAL_TIME_MODULE = 'realTimeData';
+const MODULE_NAME = 'a1Media';
+const SCRIPT_URL = 'https://linkback.contentsfeed.com/src';
+export const A1_SEG_KEY = '__a1tg';
+export const A1_AUD_KEY = 'a1gid';
+
+export const storage = getStorageManager({moduleType: MODULE_TYPE_RTD, moduleName: MODULE_NAME});
+
+/** @type {RtdSubmodule} */
+export const subModuleObj = {
+  name: MODULE_NAME,
+  init: init,
+  getBidRequestData: alterBidRequests,
+};
+
+export function getStorageData(key) {
+  let storageValue = '';
+  if (storage.getDataFromLocalStorage(key)) {
+    storageValue = storage.getDataFromLocalStorage(key);
+  } else if (storage.getCookie(key)) {
+    storageValue = storage.getCookie(key);
+  }
+  return storageValue;
+}
+
+function loadLbScript(tagname) {
+  const linkback = window.linkback = window.linkback || {};
+  if (!linkback.l) {
+    linkback.l = true;
+
+    const d = new Date();
+    const month = `0${(d.getMonth() + 1)}`.slice(-2);
+    const day = `0${d.getDate()}`.slice(-2);
+    const todaysDate = `${d.getFullYear()}${month}${day}`;
+
+    const scriptUrl = `${SCRIPT_URL}/${todaysDate}/${tagname}.min.js`;
```

Comment:
Can you add some discussion of why you need external script loaded to return segments to your documentation? It's there a reason you cannot lock this script to a version instead of this date stuff?

Reply:
The reason for needing an external script is not just to acquire data from the current page, but also to analyze accumulated data periodically. Instead of using a version, the script is loaded based on the date for a specific reason. The A1Media script has many customizable elements for the deployed site. Therefore, there is an issue where the script needs to be changed immediately upon request from the site. By loading the script based on the date, it ensures that the script is refreshed within a maximum of one day.

Reply:
Please add this to your documentation

Reply:
We added this to markdown file and https://github.com/prebid/prebid.github.io/pull/4672 too

Reply:
Thanks! Publishers generally prefer non-dynamic scripts so you don't break them with changes. We'll have to review this requested exception in committee.

Reply:
I'll go ahead and be the bad cop here -- this is simply not allowed. The [Prebid Module Rules](https://docs.prebid.org/dev-docs/module-rules.html#module-rules) clearly state:

```
The file loaded is locked to a particular version so that any upgrades must be done via Prebid pull request.
```

This is a rule specifically because many publishers do not want modules to change underneath them without notice. Your adapter is not special, and this exception will not be granted.

Here are some options:
1.  Abide by the rules and put a static version number. 
2. As your publisher partners to place your script directly on their page outside of Prebid.
3. Make the script URL a bidder-specific runtime parameter so when you urgently need to change something, you can contact the affected publisher and have them update the parameter rather than having to update the version of PBJS.
 
Note - we are fully aware that the "static version number" rule can be gamed. If you are found to be dynamically updating a supposedly static URL, there will be repercussions.

Reply:
Ok - sorry- I'll admit I thought you guys were a bid adapter at first. An RTD module is more permissive and nuanced. That rule states:
```
A Real-Time Data module may load external code if it requires publisher registration and there’s a prominent disclosure on the module documentation. The idea is that a publisher will not include the module if they don’t approve of the external code, and since they’ve registered for the service, they must approve. The text of the disclosure may differ if the vendor allows Prebid to do regular reviews of a strictly versioned proprietary library.
```
However, I don't understand what benefit you gain by putting the date in the URL. We will discuss it again internally. The proposed documented disclosure message was way too complicated. If we allow RTD modules to break this rule than publishers should not have to understand the date nuance. If anything, such a disclosure should say something simple like:

```
This module loads unverified external code that could change without warning. 
```

Reply:
As advised, we changed architecture to load external script to non-dynamic.
It just loads the script using the name passed from configuration.